Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs

```diff
@@ -0,0 +1,370 @@
+                        return null;
+                    }
+                }
+            }
+
+            for (int i = 0; i < gltfObject.extensionsUsed?.Length; i++)
+            {
+                var extensionsUsed = GetGltfExtensionObjects(jsonString, gltfObject.extensionsUsed[i]);
+
+                foreach (var extensionUsed in extensionsUsed)
+                {
+                    // TODO Update this after KHR_materials_pbrSpecularGlossiness extension is supported
+                    //if (gltfObject.extensionsUsed[i].Equals("KHR_materials_pbrSpecularGlossiness"))
+                    //{
+                    //    for (int j = 0; j < gltfObject.materials.Length; j++)
+                    //    {
+                    //        if (!string.IsNullOrEmpty(gltfObject.materials[i].name) &&
+                    //            gltfObject.materials[i].name == extensionUsed.Key)
+                    //        {
+                    //            gltfObject.materials[i].Extensions.Add(gltfObject.extensionsUsed[i], extensionUsed.Value);
+                    //            var extension = JsonUtility.FromJson<KHR_Materials_PbrSpecularGlossiness>(extensionUsed.Value);
+                    //            extension.ElementName = gltfObject.materials[i].name;
+                    //            gltfObject.RegisteredExtensions.Add(extension);
+                    //        }
+                    //    }
+                    //}
+                    //else
+                    {
+                        Debug.LogWarning($"Unsupported Extension: {gltfObject.extensionsUsed[i]}");
+                    }
+                }
+            }
+
+            var meshPrimitiveAttributes = GetGltfMeshPrimitiveAttributes(jsonString);
+
+            for (int i = 0; i < gltfObject.meshes.Length; i++)
+            {
+                for (int j = 0; j < gltfObject.meshes[i].primitives.Length; j++)
+                {
+                    gltfObject.meshes[i].primitives[j].Attributes = JsonUtility.FromJson<GltfMeshPrimitiveAttributes>(meshPrimitiveAttributes[0]);
+                    meshPrimitiveAttributes.Remove(meshPrimitiveAttributes[0]);
+                }
+            }
+
+            return gltfObject;
+        }
+
+        /// <summary>
+        /// Gets a glTF object from the provided path
+        /// </summary>
+        /// <param name="glbData">Raw glb byte data.</param>
+        /// <returns><see cref="GltfObject"/></returns>
+        /// <remarks>Returned <see cref="GltfObject"/> still needs to be initialized using <see cref="ConstructGltf.ConstructAsync"/>.</remarks>
+        public static GltfObject GetGltfObjectFromGlb(byte[] glbData)
+        {
+            const int stride = sizeof(uint);
+
+            var magicNumber = BitConverter.ToUInt32(glbData, 0);
+            var version = BitConverter.ToUInt32(glbData, stride);
+            //var length = BitConverter.ToUInt32(glbData, stride * 2);
```

Comment:
if this code isn't used/needed we should remove it

Reply:
We may actually benefit from validating the glbData payload/size given this length value.

Reply:
Correct, we should be, but I didn't have time to implement it.

Reply:
picking up this fix